packages/react-button/src/components/CompoundButton/CompoundButton.test.tsx

```diff
@@ -8,4 +9,126 @@ describe('CompoundButton', () => {
     Component: CompoundButton as React.FunctionComponent<CompoundButtonProps>,
     displayName: 'CompoundButton',
   });
+
+  describe('when rendered as a button', () => {
+    it('renders correctly', () => {
+      const result = render(<CompoundButton>This is a button</CompoundButton>);
+      const button = result.getByRole('button');
+
+      expect(button).toBeTruthy();
+      expect(button.tagName).toBe('BUTTON');
+    });
+
+    it('can be focused', () => {
+      const result = render(<CompoundButton>This is a button</CompoundButton>);
+      const button = result.getByRole('button');
+
+      expect(document.activeElement).not.toEqual(button);
+      button.focus();
+      expect(document.activeElement).toEqual(button);
+    });
+
+    it('cannot be focused when disabled has been passed to the component', () => {
+      const result = render(<CompoundButton disabled>This is a button</CompoundButton>);
+      const button = result.getByRole('button');
+
+      expect(document.activeElement).not.toEqual(button);
+      button.focus();
+      expect(document.activeElement).not.toEqual(button);
+    });
+
+    it('can be focused when disabledFocusable has been passed to the component', () => {
+      const result = render(<CompoundButton disabledFocusable>This is a button</CompoundButton>);
+      const button = result.getByRole('button');
+
+      expect(document.activeElement).not.toEqual(button);
+      button.focus();
+      expect(document.activeElement).toEqual(button);
+    });
+
+    it('can trigger a function by being clicked', () => {
+      const onClick = jest.fn();
+      const result = render(<CompoundButton onClick={onClick}>This is a button</CompoundButton>);
+
+      fireEvent.click(result.getByRole('button'));
```

Comment:
You can optionally also trigger clicks with `userEvent`. From the basic description in the docs it sounds like the main difference is userEvent also does hover events prior to the click. (The general idea of userEvent is trying to more realistically simulate interaction.)

Reply:
We should definitely be as realistic as possible with our tests. Maybe this should also be a Cypress test since that will be an actual click in a browser?

Reply:
I'll keep that in mind for a separate PR and merge this one as is 😄

Reply:
I don't think this test needs to be in Cypress. Clicks can be simulated realistically enough for this test's needs in jest/jsdom (and these tests are quicker to run than cypress tests). Cypress is most useful for things like focus which aren't realistically simulated by jsdom.